session/txn.go

```diff
@@ -181,20 +204,20 @@ func (txn *LazyTxn) GoString() string {
 
 func (txn *LazyTxn) changeInvalidToValid(kvTxn kv.Transaction) {
 	txn.Transaction = kvTxn
-	atomic.StoreInt32(&txn.State, txninfo.TxnRunningNormal)
-	atomic.StoreUint64(&txn.infoStartTS, kvTxn.StartTS())
 	txn.initStmtBuf()
-	atomic.StoreUint64(&txn.EntriesCount, uint64(txn.Transaction.Len()))
-	atomic.StoreUint64(&txn.EntriesSize, uint64(txn.Transaction.Size()))
+	txn.recreateTxnInfo(
```

Comment:
Why atomic? when will the `changeInvalidToValid()` function been called concurrently? @MyonKeminta

Reply:
It should not be called concurrently, but the content of the `txnInfo` may be retrieved by another thread outside the session.